l1-contracts/contracts/state-transition/ChainTypeManager.sol

```diff
@@ -229,6 +232,12 @@ contract ChainTypeManager is IChainTypeManager, ReentrancyGuard, Ownable2StepUpg
         emit NewValidatorTimelock(oldValidatorTimelock, _validatorTimelock);
     }
 
+    /// @dev set ServerNotifier.
+    /// @param _serverNotifier the new serverNotifier address
+    function setServerNotifier(address _serverNotifier) external onlyOwner {
+        serverNotifierAddress = _serverNotifier;
```

Comment:
Don't forget to add `event` :)

Reply:
`onlyAdmin`?

Reply:
I used the same restriction, as for validator timelock